.github/workflows/magma-build-3rd-party.yml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Magma Build & Publish 3rd Party Dependencies
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  pull_request:
+    types: [ opened, reopened, synchronize ]
```

Comment:
Building and publishing 3rd party dependencies on every PR sync seems overkill to me.

Reply:
It is only on every PR with code change of the respective parts. See lines 26ff below.

Reply:
Ah yes, of course. But wouldn't uploading would fail because the secrets aren't available in PR context?

Reply:
Good point. This depends where your branch resides. For this PR it is obviously working. For forked branches it probably won't.

Reply:
Safeguard against missing credentials added.